packages/kbn-optimizer/src/common/disallowed_syntax_plugin/disallowed_syntax.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import estree from 'estree';
+
+export interface DisallowedSyntaxCheck {
+  name: string;
+  nodeType: estree.Node['type'] | Array<estree.Node['type']>;
+  test?: (n: any) => boolean | void;
+}
+
+export const checks: DisallowedSyntaxCheck[] = [
+  /**
+   * es2015
+   */
+  // https://github.com/estree/estree/blob/master/es2015.md#functions
+  {
+    name: '[es2015] generator function',
+    nodeType: ['FunctionDeclaration', 'FunctionExpression'],
+    test: (n: estree.FunctionDeclaration | estree.FunctionExpression) => !!n.generator,
```

Comment:
I would be surprised if we were to be the first project to have such needs. Isn't there no oss tools performing such checks atm?

Reply:
I looked around and didn't find anything, but I would be shocked if we could find something that would integrate with the webpack parser so tightly in order to avoid parsing the bundles all over again.

Reply:
Yea, from a quick glance, every webpack plugin integration (ie https://github.com/bitjourney/check-es-version-webpack-plugin/blob/master/index.js) are also using acorn at the end and parses the whole files again. None are doing the fine-grained checks you implemented though.